src/pgduckdb_types.cpp

```diff
@@ -216,6 +235,20 @@ ConvertBinaryDatum(const duckdb::Value &value) {
 inline Datum
 ConvertDateDatum(const duckdb::Value &value) {
 	duckdb::date_t date = value.GetValue<duckdb::date_t>();
+	if (!ValidDate(date))
+		throw duckdb::OutOfRangeException("The value should be between min and max value (%s <-> %s)",
+		                                  duckdb::Date::ToString(pgduckdb::PGDUCKDB_PG_MIN_DATE_VALUE),
+		                                  duckdb::Date::ToString(pgduckdb::PGDUCKDB_PG_MAX_DATE_VALUE));
+
+	// Special Handling for +/-infinity date values
+	if (!duckdb::Date::IsFinite(date)) {
+		// -infinity value is different for PG date
+		if (date == duckdb::date_t::ninfinity())
+			return date.days - 1;
+		else
+			return date.days;
+	}
```

Comment:
Instead of doing this `-1` calculation let's use the constants provided by PG/DuckDB. Same for other places in the new code.
```suggestion
	if (date == duckdb::date_t::ninfinity())
		return DATEVAL_NOEND;
	else if (date == duckdb::date_t::infinity())
		return DATEVAL_NOBEGIN;
```